File: frontend/src/types/payment.ts
export type PaymentProvider = "STRIPE";

export type PaymentStatus =
  | "PENDING"
  | "SUCCEEDED"
  | "FAILED"
  | "CANCELLED"
  | "REFUNDED"
  | "PARTIALLY_REFUNDED";

export interface PaymentResponseDto {
  id: string;
  orderId: number;
  provider: PaymentProvider;
  status: PaymentStatus;
  providerPaymentIntentId?: string | null;
  providerChargeId?: string | null;
  amount: number;
  currency: string;
  failureCode?: string | null;
  failureMessage?: string | null;
  createdAt: string;
  updatedAt: string;
  paidAt?: string | null;
}

export type RefundStatus = "PENDING" | "SUCCEEDED" | "FAILED";

export interface RefundResponseDto {
  id: string;
  paymentId: string;
  orderId: number;
  orderItemId?: number | null;
  amount: number;
  currency: string;
  reason?: string | null;
  quantity?: number | null;
  status: RefundStatus;
  stripeRefundId?: string | null;
  createdAt: string;
  succeededAt?: string | null;
}

export interface StripePaymentIntentResponse {
  paymentId: string;
  orderId: number;
  clientSecret?: string | null;
  amount: number;
  currency: string;
  status: PaymentStatus;
}

File: backend/src/main/java/com/ecommercestore/backend/payment/Payment.java
package com.ecommercestore.backend.payment;

import com.ecommercestore.backend.order.Order;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentProvider provider;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus status;

    private String providerPaymentIntentId;

    private String providerChargeId;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(nullable = false)
    private String currency;

    private String failureCode;

    @Column(length = 2000)
    private String failureMessage;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;

    private Instant paidAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;

        if (provider == null) {
            provider = PaymentProvider.STRIPE;
        }

        if (status == null) {
            status = PaymentStatus.PENDING;
        }

        if (currency == null) {
            currency = "EUR";
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
